eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/MutableListFactoryImpl.java

```diff
@@ -36,14 +36,16 @@
         return this.empty();
     }
 
+    @SafeVarargs
     @Override
-    public <T> MutableList<T> of(T... items)
+    public final <T> MutableList<T> of(T... items)
     {
         return this.with(items);
     }
 
+    @SafeVarargs
     @Override
-    public <T> MutableList<T> with(T... items)
+    public final <T> MutableList<T> with(T... items)
     {
         return FastList.newListWith(items);
```

Comment:
My hypothesis is that the immutable factories are in fact safe, and that the mutable factories are not safe. I am most confident that this is a good example of unsafe varargs. I would ask that we reason through what we think is safe or not, rather than go with my guess that the immutable factories are ok.

Reply:
@motlin I had the same thought when implementing that. I agree that we should reason through  👍

Reply:
And I do agree that the whole class should be final because It makes sense and I don't expect anybody to override them. ( While I debated a lot with myself about this, I am not finding a proper use case.)  The one obvious thing is that users may need to modify some aspects of the underlying data structure, they can extend.